src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Attr.java

```diff
@@ -2418,6 +2421,40 @@ public void visitApply(JCMethodInvocation tree) {
 
             chk.checkRefTypes(tree.typeargs, typeargtypes);
 
+            if (localEnv.info.lint.isEnabled(LintCategory.SYNCHRONIZE)) {
```

Comment:
Looks like you're checking calls to wait/notify? I don't think that's necessary—there will already be a warning at the `synchronized` statement; if that doesn't happen, it's malformed code that will fail at runtime (attempting to wait without owning the lock). In practice, I think we just end up with lots of redundant warnings.

Reply:
Sounds good. I agree this is redundant and noisy.